File: VCSJones.FiddlerCertGen/PublicKeyInfo.cs
using System;
using System.Runtime.InteropServices;
using VCSJones.FiddlerCertGen.Interop;

namespace VCSJones.FiddlerCertGen
{
    public class PublicKeyInfo : IDisposable
    {
        private readonly CERT_PUBLIC_KEY_INFO _publicKey;
        private readonly IntPtr _buffer;

        internal PublicKeyInfo(IntPtr buffer)
        {
            _publicKey = (CERT_PUBLIC_KEY_INFO)Marshal.PtrToStructure(buffer, typeof(CERT_PUBLIC_KEY_INFO));
            _buffer = buffer;
        }

        internal CERT_PUBLIC_KEY_INFO PublicKey => _publicKey;

        public byte[] Key
        {
            get
            {
                var key = new byte[_publicKey.PublicKey.cbData];
                Marshal.Copy(_publicKey.PublicKey.pbData, key, 0, key.Length);
                return key;
            }
        }

        public void Dispose()
        {
            GC.SuppressFinalize(this);
            Marshal.FreeHGlobal(_buffer);

        }

        ~PublicKeyInfo()
        {
            Dispose();
        }
    }
}

File: VCSJones.FidderCertGenFS/PublicKeyInfo.fs
namespace VCSJones.FiddlerCertGen
open System
open System.Runtime.InteropServices

type PublicKeyInfo(buffer : GlobalBufferSafeHandle) =
    let publicKey = Marshal.PtrToStructure(buffer.DangerousGetHandle(), typeof<CERT_PUBLIC_KEY_INFO>) :?> CERT_PUBLIC_KEY_INFO

    member internal this.PublicKey = publicKey

    member this.Key = 
        let key = Array.zeroCreate<byte> (int publicKey.PubicKey.cbData)
        Marshal.Copy(publicKey.PubicKey.pbData, key, 0, key.Length)
        key

    override this.Finalize() = 
        (this :> IDisposable).Dispose()

    interface IDisposable with
        member this.Dispose() =
            GC.SuppressFinalize(this)
            buffer.Dispose()